pkg/build/dockerfile-generater.go

```diff
@@ -22,6 +22,7 @@ type DockerfileGenerator struct {
 	*manifest.Manifest
 	*templates.Templates
 	mixin.MixinProvider
+	hasBuildArg bool
```

Comment:
`hasBuildArg` can be defined as a local variable inside of `readAndValidateDockerfile` since it is set and used only in that function.

Reply:
@carolynvs made the changes :)